include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -274,7 +274,7 @@
 //'present' macros
 // shift_left, shift_right; GCC 10; VS 2019 16.1
 #define _ONEDPL_CPP20_SHIFT_LEFT_RIGHT_PRESENT                                                                         \
-    (__cplusplus >= 202002L && ((_GLIBCXX_RELEASE >= 10) || (_MSC_VER >= 1921)))
+    ((__cplusplus >= 202002L || _MSVC_LANG >= 202002L) && (_GLIBCXX_RELEASE >= 10 || _MSC_VER >= 1921))
```

Comment:
I believe we can use `_ONEDPL_cplusplus` macro here as well (without adding an include of `<common_config.h>. See other comments for the explanation`)

Reply:
Right now we already have two definition's of our CPP version macros:
1) https://github.com/oneapi-src/oneDPL/pull/678/files#diff-4c32e30dceaeba2ff83a44fecd56fb4bfe504e60b3f0e2bed77a74abc0b54176
```
#ifndef _MSVC_LANG
#    define _ONEDPL_cplusplus __cplusplus
#else
#    define _ONEDPL_cplusplus _MSVC_LANG
#endif
```
2) https://github.com/oneapi-src/oneDPL/pull/678/files#diff-d96b6ea8eddec6ed0f860b2658df3af0306f3583917866e78ec8d54f8703f986
```
#ifndef _MSVC_LANG
#    define _ONEDPL_TEST_cplusplus __cplusplus
#else
#    define _ONEDPL_TEST_cplusplus _MSVC_LANG
#endif
```

@rarutyun you propose to make in this fine the third new define of something like this?

Reply:
fixed

Reply:
fixed